### PR TITLE
Add ts types generation to static checks

### PR DESCRIPTION
**Pull request by pierrejeambrun**
Following https://github.com/apache/airflow/pull/25123, allowing us to auto-generate typescript types for the api, here is a pre-commit hook to ensure that these types stay in sync with the OpenAPI spec.

Modifying the `openapi/v1.yaml` will trigger the type generation. (then could eventually break ESlint hook when attempting again to commit, in case of incompatible changes)

### CODE REVIEW

**Review by bbovenzi — APPROVED** [review]
looking great!

**Comment by bbovenzi** [comment]
I merged my PR. Want to rebase and update the PR title and description?

**Comment by pierrejeambrun** [comment]
The PR has been updated, we should be good to go!

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
One small comment - it woudl be - I think - much better to merge this one with `compile-www-assets` - then same node env will be used and single yarn install command would be needed

**Comment by potiuk** [comment]
Those are triggered by different source files - but I think overall it would be more efficient.

**Comment by pierrejeambrun** [comment]
The `compile-www-assets` has a `manual` confined stage, should I extend this with commit ?

**Comment by potiuk** [comment]
Right.... Forgot about it 🤦   But the "lint" one does not :). This what Is actually better "typescript compile lint" sems like good idea

**Comment by potiuk** [comment]
On the other hand. isolation is better in this case rather than optimisation.

**Comment by pierrejeambrun** [comment]
Following your suggestions, here is another option that merges the `generate-api-types` hook into `lint-javascript` (+ renaming reflecting this change).

Let me know what version you prefer :)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -834,12 +834,12 @@ repos:
         entry: ./scripts/ci/pre_commit/pre_commit_ui_lint.py
         pass_filenames: false
         additional_dependencies: ['yarn@1.22.19']
-      - id: lint-javascript
+      - id: ts-compile-and-lint-javascript
         name: ESLint against current UI JavaScript files
```
Comment (potiuk) [inline .pre-commit-config.yaml]: Let's change the description too.

**Comment by potiuk** [comment]
I thin this one is better. In VAST majority of  cases running 'generate-api-types' will produce the same file. 

Unfortunately - unlike in Python, pre-commit does not reuse node environments, so the less of them, the better for time/cache restore and creation etc:

```
⌁ [jarek:~/.cache/pre-commit] % echo */*/lib/node_modules | xargs du -h -s
 21M	repocw333njc/node_env-18.6.0/lib/node_modules
 48M	repoltxx29sq/node_env-18.6.0/lib/node_modules
 19M	repowd9np_qb/node_env-18.6.0/lib/node_modules
 20M	repoy6z0rp4o/node_env-16.15.1/lib/node_modules
```

**Comment by pierrejeambrun** [comment]
Ok great thanks for the details :)

**Comment by potiuk** [comment]
BTW. Is it possible @pierrejeambrun - you also remove all the node versions 18.6.0 in all node pre-commits and only update the version in the "top-level" default_language_version? I just realized that we have it by looking at those cached envs.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (pierrejeambrun) [reply .pre-commit-config.yaml]: Yes I didn't change it cause we are limited in length :(, trying to find something better

Comment (potiuk) [reply .pre-commit-config.yaml]: `Generate and lint UI Javascript files`  ?

**Comment by Kinceo1** [comment]
King

File: .pre-commit-config.yaml (hunk 1 above)
Comment (pierrejeambrun) [reply .pre-commit-config.yaml]: I went for `TS types generation and ESLint against current UI files` (that is bellow < 70char)  I wanted to remove the reference to Javascript as it is not really accurate anymore

**Comment by pierrejeambrun** [comment]
Done, using the default node version and updated it to `18.6.0`

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Closed/reopened to rebuild as there was temp dockerhub failure.

**Comment by pierrejeambrun** [comment]
> Closed/reopened to rebuild as there was temp dockerhub failure.

Thanks (didn't know that trick :p) I'm use to amending and repushing haha

**Comment by potiuk** [comment]
> > Closed/reopened to rebuild as there was temp dockerhub failure.
> 
> Thanks,
> 
> I didn't know that trick, I'm used to amending and repushing 😂

That's what I do with my changes too :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**